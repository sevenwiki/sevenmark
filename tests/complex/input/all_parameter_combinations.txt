{{{ #style="color:red; font-weight:bold" #size="18px" #color="blue" #bgcolor="yellow" #opacity="0.8"
Maximum styled text with all parameters
}}}

{{{#fold #style="border:2px dashed green"
[[#size="14px" #color="purple" || **Styled title** with multiple parameters]]
[[#bgcolor="lightblue" #opacity="0.9" #style="padding:10px" || Content with background and opacity]]
}}}

{{{#code #lang="javascript" #style="background:black; color:lime" #size="12px"
console.log("Styled code block");
function test() { return 42; }
}}}