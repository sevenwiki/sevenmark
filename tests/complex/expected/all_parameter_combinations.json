[
  {
    "StyledElement": {
      "location": {
        "start": 0,
        "end": 144
      },
      "common_style": {
        "style": [
          {
            "Text": {
              "location": {
                "start": 12,
                "end": 39
              },
              "content": "color:red; font-weight:bold"
            }
          }
        ],
        "size": [
          {
            "Text": {
              "location": {
                "start": 48,
                "end": 52
              },
              "content": "18px"
            }
          }
        ],
        "color": [
          {
            "Text": {
              "location": {
                "start": 62,
                "end": 66
              },
              "content": "blue"
            }
          }
        ],
        "bg_color": [
          {
            "Text": {
              "location": {
                "start": 78,
                "end": 84
              },
              "content": "yellow"
            }
          }
        ],
        "opacity": [
          {
            "Text": {
              "location": {
                "start": 96,
                "end": 99
              },
              "content": "0.8"
            }
          }
        ]
      },
      "content": [
        {
          "Text": {
            "location": {
              "start": 101,
              "end": 140
            },
            "content": "Maximum styled text with all parameters"
          }
        },
        {
          "Text": {
            "location": {
              "start": 140,
              "end": 141
            },
            "content": "\n"
          }
        }
      ]
    }
  },
  {
    "Text": {
      "location": {
        "start": 144,
        "end": 145
      },
      "content": "\n"
    }
  },
  {
    "Text": {
      "location": {
        "start": 145,
        "end": 146
      },
      "content": "\n"
    }
  },
  {
    "FoldElement": {
      "location": {
        "start": 146,
        "end": 370
      },
      "common_style": {
        "style": [
          {
            "Text": {
              "location": {
                "start": 163,
                "end": 186
              },
              "content": "border:2px dashed green"
            }
          }
        ]
      },
      "content": [
        {
          "common_style": {
            "size": [
              {
                "Text": {
                  "location": {
                    "start": 197,
                    "end": 201
                  },
                  "content": "14px"
                }
              }
            ],
            "color": [
              {
                "Text": {
                  "location": {
                    "start": 211,
                    "end": 217
                  },
                  "content": "purple"
                }
              }
            ]
          },
          "content": [
            {
              "Bold": {
                "location": {
                  "start": 222,
                  "end": 238
                },
                "content": [
                  {
                    "Text": {
                      "location": {
                        "start": 224,
                        "end": 236
                      },
                      "content": "Styled title"
                    }
                  }
                ]
              }
            },
            {
              "Text": {
                "location": {
                  "start": 238,
                  "end": 263
                },
                "content": " with multiple parameters"
              }
            }
          ]
        },
        {
          "common_style": {
            "style": [
              {
                "Text": {
                  "location": {
                    "start": 312,
                    "end": 324
                  },
                  "content": "padding:10px"
                }
              }
            ],
            "bg_color": [
              {
                "Text": {
                  "location": {
                    "start": 278,
                    "end": 287
                  },
                  "content": "lightblue"
                }
              }
            ],
            "opacity": [
              {
                "Text": {
                  "location": {
                    "start": 299,
                    "end": 302
                  },
                  "content": "0.9"
                }
              }
            ]
          },
          "content": [
            {
              "Text": {
                "location": {
                  "start": 329,
                  "end": 364
                },
                "content": "Content with background and opacity"
              }
            }
          ]
        }
      ]
    }
  },
  {
    "Text": {
      "location": {
        "start": 370,
        "end": 371
      },
      "content": "\n"
    }
  },
  {
    "Text": {
      "location": {
        "start": 371,
        "end": 372
      },
      "content": "\n"
    }
  },
  {
    "CodeElement": {
      "location": {
        "start": 372,
        "end": 519
      },
      "language": [
        {
          "Text": {
            "location": {
              "start": 388,
              "end": 398
            },
            "content": "javascript"
          }
        }
      ],
      "content": [
        {
          "Text": {
            "location": {
              "start": 451,
              "end": 501
            },
            "content": "console.log(\"Styled code block\");\nfunction test() "
          }
        },
        {
          "Text": {
            "location": {
              "start": 501,
              "end": 502
            },
            "content": "{"
          }
        },
        {
          "Text": {
            "location": {
              "start": 502,
              "end": 514
            },
            "content": " return 42; "
          }
        },
        {
          "Text": {
            "location": {
              "start": 514,
              "end": 515
            },
            "content": "}"
          }
        },
        {
          "Text": {
            "location": {
              "start": 515,
              "end": 516
            },
            "content": "\n"
          }
        }
      ]
    }
  }
]